lib/veri_traci.mli

```diff
@@ -0,0 +1,30 @@
+open Bap.Std
+open Bap_traces.Std
+open Bil.Result
+
+type event = Trace.event
+
+class context : arch -> object('s)
+    inherit Bili.context
+    method update_event : event -> 's
+    method arch: arch
+    method mem : var
+  end
+
+class ['a] t : object('s)
+  constraint 'a = #context
+  inherit ['a] Bili.t
+  method eval_event : event -> 'a u
```

Comment:
you should add `method eval_trace : trace -> 'a u`, that will run the whole sequence of events. You can use `event seq -> 'a u` if you find it more suitable.